File: balanced_brackets/balanced_brackets.py
def is_balanced(brackets):
    stack = []
    opening_brackets = ["(", "[", "{"]
    match = {")": "(", "]": "[", "}": "{"}

    for b in brackets:
        if b in opening_brackets:
            stack.append(b)
        if b in match:
            if not stack or stack.pop() != match[b]:
                return False

    if stack:
        return False

    return True

File: balanced_brackets/BalancedBrackets.java
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.Map;

class BalancedBrackets {

    private final Deque<Character> stack = new ArrayDeque<>();
    private final Map<Character, Character> bracketMap = Map.of(
            '(', ')',
            '[', ']',
            '{', '}');

    public Boolean isBalanced(String brackets) {
        for (char c : brackets.toCharArray()) {
            if (isOpeningBracket(c))
                stack.push(c);
            else if (stack.isEmpty() || bracketMap.get(stack.pop()) != c)
                return false;
        }

        return stack.isEmpty();
    }

    private Boolean isOpeningBracket(Character chr) {
        return bracketMap.containsKey(chr);
    }

}
